src/iidm/Extendable.cpp

```diff
@@ -28,6 +28,8 @@ Extendable::Extendable(Extendable&& extendable) noexcept :
 
 void Extendable::addExtension(std::unique_ptr<Extension>&& extension) {
     extension->setExtendable(*this);
+    m_extensionsByName.erase(extension->getName());
+    m_extensionsByType.erase(extension->getType());
```

Comment:
To be discussed: I wonder if it would be better to use the `operator[]` that will create a new entry if the key doesn't exist and replace the value otherwise.

Another option would be to create a utility function in `map.hpp` to provide such method.